src/components/common/quantity/Quantity.tsx

```diff
@@ -1,37 +1,116 @@
-import React from "react";
+import React, { useEffect, useState, useCallback } from "react";
 import {
   AutoComplete,
   TextField,
 } from "@madie/madie-design-system/dist/react/";
 import "twin.macro";
 import "styled-components/macro";
+import { CQL } from "cqm-models";
+import * as ucum from "@lhncbc/ucum-lhc";
 
 export interface QuantityProps {
-  quantityValue: number;
-  quantityUnit: any;
-  handleQuantityValueChange: Function;
-  handleQuantityUnitChange: Function;
-  options?: any;
+  quantity: CQL.Quantity;
+  handleQuantityChange?: Function;
   canEdit: boolean;
   label: string;
 }
 
+export interface UcumOption {
+  label: string;
+  value: ucum;
+}
+
 const Quantity = ({
```

Comment:
could we rename this to `QuantityInput` so there's no naming overlap with the datatypes coming from cql-execution?

Reply:
Ok